### PR TITLE
Add org.kde.sweeper

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.
> Utility to clear caches
- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://invent.kde.org/teams/flathub/issues/-/issues/14
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://invent.kde.org/utilities/sweeper/-/merge_requests/18

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.kde.sweeper

File: org.kde.sweeper.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    "sdk": "org.kde.Sdk",
+    "rename-icon": "sweeper",
+    "command": "sweeper",
+    "finish-args": [
+        "--device=dri",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--talk-name=org.kde.sweeper",
+        "--filesystem=xdg-cache"
```
Comment (hfiguiere) [inline org.kde.sweeper.json:14]: why would it have this permission? it's the build failure.

File: org.kde.sweeper.json [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    "runtime-version": "6.7",
+    "sdk": "org.kde.Sdk",
+    "rename-icon": "sweeper",
+    "command": "sweeper",
+    "finish-args": [
+        "--device=dri",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--talk-name=org.kde.sweeper",
```
Comment (hfiguiere) [inline org.kde.sweeper.json:13]:
```suggestion
```

this is granted by default.

File: appstream.patch [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+---
+ org.kde.sweeper.appdata.xml | 2 ++
+ 1 file changed, 2 insertions(+)
+
+diff --git a/org.kde.sweeper.appdata.xml b/org.kde.sweeper.appdata.xml
+index d2dcd0a..f75b3fc 100644
+--- a/org.kde.sweeper.appdata.xml
++++ b/org.kde.sweeper.appdata.xml
+@@ -3,6 +3,7 @@
+   <id>org.kde.sweeper.desktop</id>
```
Comment (hfiguiere) [inline appstream.patch:13]: @bbhtt does this `id` need to be changed?

File: org.kde.sweeper.json (hunk 1 above)
Comment (Mailaender) [reply org.kde.sweeper.json:14]: It needs access to `~/.cache`.

**Comment by Mailaender** [comment]
bot, build org.kde.sweeper

File: appstream.patch (hunk 3 above)
Comment (Mailaender) [reply appstream.patch:13]: Yes https://invent.kde.org/teams/flathub/issues/-/issues/28#error-appstream-id-mismatch-flatpak-id

**Comment by Mailaender** [comment]
bot, build org.kde.sweeper

File: appstream.patch [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+diff --git a/org.kde.sweeper.appdata.xml b/org.kde.sweeper.appdata.xml
+index f75b3fc..94fd0ca 100644
+--- a/org.kde.sweeper.appdata.xml
++++ b/org.kde.sweeper.appdata.xml
+@@ -1,6 +1,6 @@
+ <?xml version="1.0" encoding="utf-8"?>
+ <component type="desktop">
+-  <id>org.kde.sweeper.desktop</id>
++  <id>org.kde.sweeper</id>
```
Comment (bbhtt) [inline appstream.patch:60]: You have to add a provides tag. https://docs.flathub.org/docs/for-app-authors/metainfo-guidelines/#renaming-id-tag

Comment (Mailaender) [reply appstream.patch:60]: This is not stated in https://invent.kde.org/teams/flathub/issues/-/issues/28#error-appstream-id-mismatch-flatpak-id

**Comment by Mailaender** [comment]
bot, build org.kde.sweeper

File: appstream.patch (hunk 4 above)
Comment (bbhtt) [reply appstream.patch:60]: Needs to be documented there too.

File: appstream.patch [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+   <metadata_license>FSFAP</metadata_license>
+   <project_license>LGPL-2.1+</project_license>
+   <developer_name translate="no">KDE</developer_name>
+@@ -179,4 +179,10 @@
+     <release version="23.08.5" date="2024-02-15"/>
+   </releases>
+   <launchable type="desktop-id">org.kde.sweeper.desktop</launchable>
++  <provides>
++    <id>org.kde.sweeper.desktop</id>
++  </provides>
```
Comment (bbhtt) [inline appstream.patch:70]: Looks like there is already a provides tag in metainfo file somewhere.

Comment (Mailaender) [reply appstream.patch]: Sorry, I am not very good at this patch juggling.

**Comment by Mailaender** [comment]
bot, build org.kde.sweeper

**Comment by bbhtt** [comment]
Looks good from my side waiting for a comment from KDE team.

**Comment by travier** [comment]
Hum, doesn't this need full RW access to the home directory and all dot folder to be useful?

**Comment by Mailaender** [comment]
I forgot to add a replacement for `xdg-cache` which was removed during review.

**Comment by Mailaender** [comment]
bot, build org.kde.sweeper

**Comment by travier** [comment]
bot, build org.kde.sweeper

**Comment by travier** [comment]
Manifest LGTM and the app runs.

However I've not tried it yet as I don't know exactly what it removes and it has full home access so I will have to make backups before just to be safe here or run that in a VM.

**Comment by Mailaender** [comment]
I am also unsure if it works at all because of the sandbox restrictions. Sounds like → `beta` to me.

**Comment by Justinzobel** [comment]
Ping.

**Comment by travier** [comment]
@Justinzobel We need someone to do the testing for this one.

**Comment by Justinzobel** [comment]
I've never used it so probably not a good test candidate. Was just trying to see if we can move it along.

**Comment by RacerBG** [comment]
@travier The application itself is working however something doesn't seem right.

1. The "General" settings are cleaned as they should in my case.
2. The "Web Browsing" though is a different beast. I assume that the problem might be that I'm using Firefox as a flatpak application. I tried to clean only certain stuff because I don't want to nuke my browsing history and cookies in the process. :smile: 
2.1. Favorite Icons and Form Completion Entries are claimed to be cleaned by Sweeper however they are not as far as I can tell (this applies more for the latter because I can't prove that for the former).
2.2. The Web Cache cleaning is definitely failing both in Sweeper and as shown in the browser. I used to have around 1.2 GB cache and it's still the same.

OS: Debian 12 Bookworm
Desktop: KDE 5.27.2
Flatpak: 1.14.10
Firefox (as a flatpak package): 131.0

**Comment by travier** [comment]
So it looks like this works at least a bit but might need some development work. Maybe a good fit for beta?

**Comment by RacerBG** [comment]
> So it looks like this works at least a bit but might need some development work. Maybe a good fit for beta?

Beta sounds good to me. At least Sweeper will be "out there" and more people could test it and report any potential issues. But it's far from ready for anything else.

**Pull request closed without merging**